src/App/Fossa/Analyze.hs

```diff
@@ -258,6 +258,7 @@ analyze cfg = Diag.context "fossa-analyze" $ do
         else Diag.context "fossa-deps" . runStickyLogger SevInfo $ analyzeFossaDepsFile basedir apiOpts
   let additionalSourceUnits :: [SourceUnit]
       additionalSourceUnits = mapMaybe (join . resultToMaybe) [manualSrcUnits, vsiResults, binarySearchResults]
+  traverse_ (Diag.flushLogs SevError SevWarn) [vsiResults, binarySearchResults, manualSrcUnits]
```

Comment:
Looks like the best way is to do this:
```haskell
additionalSourceUnits <- mapMaybe join <$> traverse (recover . rethrow) [manualSrcUnits, vsiResults, binarySearchResults]
```

Reply:
We lose the ability to set the severity with this approach, e.g. show warnings when only when --debug is provided

Reply:
Logging is useful, but if we log things in the middle of execution, then log a bunch more execution, users will likely ignore it, unless we fail outright.  This approach has the advantage of reporting it at the end, with all other warnings.  Since the user doesn't have to watch the execution, or scroll up, it's less likely that they'll miss this.

A common theme from zendesk is "The logs clearly state what went wrong, and it's either user-error or user-resolvable, but the user didn't trudge through the whole log to find it."  By surfacing these types of warnings at the end, people are more likely to read it, and those ZD tickets get created less.

Reply:
Yeah - all errors/warns are emitted in tmp/scan summary file as part of this: https://github.com/fossas/team-analysis/issues/856.